gradle/validation/rat-sources.gradle

```diff
@@ -0,0 +1,266 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import groovy.xml.NamespaceBuilder
+
+configure(rootProject) {
+    configurations {
+        rat
+    }
+
+    dependencies {
+        rat "org.apache.rat:apache-rat"
+    }
+}
+
+allprojects {
+    task("rat", type: RatTask) {
+        group = 'Verification'
+        description = 'Runs Apache Rat checks.'
+    }
+
+    if (project == rootProject) {
+        rat {
+            includes += [
+                "buildSrc/**/*.java",
+                "lucene/tools/forbiddenApis/**",
+                "lucene/tools/prettify/**",
+                // "dev-tools/**"
+            ]
+            excludes += [
+                // Unclear if this needs ASF header, depends on how much was copied from ElasticSearch
+                "**/ErrorReportingTestListener.java"
+            ]
+        }
+    }
+
+    if (project.path == ":lucene:analysis:common") {
+        rat {
+            srcExcludes += [
+                "**/*.aff",
+                "**/*.dic",
+                "**/charfilter/*.htm*",
+                "**/*LuceneResourcesWikiPage.html"
+            ]
+        }
+    }
+
+    if (project.path == ":lucene:analysis:kuromoji") {
+        rat {
+            srcExcludes += [
+                // whether rat detects this as binary or not is platform dependent?!
```

Comment:
This is one of the reasons I'd rather have our own check - we don't have to rely on a black box (or look inside rat's code to figure out what it does).